api/apps/api/src/modules/scenarios/delete-scenario/delete-scenario.handler.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { GeoFeature } from '@marxan-api/modules/geo-features/geo-feature.api.entity';
+import { FakeLogger } from '@marxan-api/utils/__mocks__/fake-logger';
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { Injectable, Logger } from '@nestjs/common';
+import { CqrsModule, EventBus, IEvent } from '@nestjs/cqrs';
+import { Test } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { isLeft, isRight } from 'fp-ts/lib/Either';
+import { v4 } from 'uuid';
+import { ScenarioDeleted } from '../events/scenario-deleted.event';
+import { Scenario } from '../scenario.api.entity';
+import {
+  DeleteScenario,
+  deleteScenarioFailed,
+  DeleteScenarioResponse,
+} from './delete-scenario.command';
+import { DeleteScenarioHandler } from './delete-scenario.handler';
+
+let fixtures: FixtureType<typeof getFixtures>;
+
+beforeEach(async () => {
+  fixtures = await getFixtures();
+});
+
+it('deletes a scenario and emits a ScenarioDeleted event', async () => {
+  const scenarioId = fixtures.GivenScenarioExits();
+
+  const result = await fixtures.WhenAScenarioIsDeleted(scenarioId);
+
+  fixtures.ThenTrueIsReturned(result);
+  await fixtures.ThenScenarioIsDeleted(scenarioId);
+  fixtures.ThenAScenarioDeletedEventIsEmitted(scenarioId);
+});
+
+it('fails to delete a scenario ', async () => {
+  const scenarioId = fixtures.GivenScenarioExits();
+  fixtures.GivenDeleteOperationFails();
+
+  const result = await fixtures.WhenAScenarioIsDeleted(scenarioId);
+
+  fixtures.ThenDeleteScenarioFailedIsReturned(result);
```

Comment:
very picky detail - I'd say this is a `when` (when a scenario deletion is requested and it fails...)